browser/src/layer/tile/CalcTileLayer.js

```diff
@@ -255,7 +255,7 @@ L.CalcTileLayer = L.CanvasTileLayer.extend({
 
 		this._map.fire('scrolllimits', newSizePx.clone());
 
-		if (limitWidth || limitHeight || extendedLimit)
+		if (!this._syncTileContainerSize() && (limitWidth || limitHeight || extendedLimit))
```

Comment:
This doesn't seem right to me, but perhaps I've misunderstood. Shouldn't these be reversed? It sounds like `_syncTileContainerSize` only needs to be called in the case that the unchanged clause is true - but this will now always call it.

Reply:
It returns true if it calls requestReDraw. So if it false, we can reDraw.

Reply:
That isn't quite what I meant - I don't think we should be calling `sync_tileContainerSize` unconditionally here, surely it only makes sense to call it when `limitWidth || limitHeight || extendedLimit` is true? It feels to me like the correct if statement here is `if ((limitWidth || limitHeight || extendedLimit) && !this._syncTileContainerSize())`. If it isn't, I'd like to know why you think differently so I can improve my understanding of what's happening here.

Reply:
I'd even go as far as to say actually that relying on the side-effect of a multi-clause if statement is bad practice and the if statement should be unchanged, but the contents of the block should be `if (!this._syncTileContainerSize()) app.sectionContainer.requestReDraw();`

Reply:
Restricted document size may result in different view size. That's why we need to call syncTilecontainersize there. restrictDocumentSize calls requestReDraw if the allowed borders change. syncTileContainerSize calls requestReDraw if the size of the HTML elements change. We need those 2 checks i guess. Multi-clause if statement may or may not be a bad practice. That's arguable. I assume you want a rafactoring here, that's not the intention of this commit and i don't think it would be straightforward. I guess we can work on it later.

Reply:
That's still not quite what I mean - if I understand this function correctly (which I may not :)), the view size is only changed if `limitWidth || limitHeight || extendedLimit` is `true`. If all of those are false, the view hasn't changed size, so there should be no need to call `_syncTileContainerSize` (which is expensive) - is this not the case?

Reply:
newDocWidth is re-set regardless of the value of  "limitWidth" and same is true for newDocHeight. That code got a little chaotic in time i think. My brain stopped functioning there and just called the function.

There is a "shouldRestrict" variable and we can assume that if the size is changed, syncTileContainerSize will request a redraw. But it requests reDraw if only the size is higher than before.

So we turn back to first state and rely on "limitWidth ||...." etc checks.

This code is really hard to follow. Really sorry if i am insisting unnecessarily. I think we strongly need to refactor that part.

Reply:
I think we need to refactor syncTileContainer size and shouldn't let it set the CSS variables unnecessarily. I see your point, that's expensive. Then i guess we can add some checks into that function.

Reply:
Ok, in that case I think I was misunderstanding this function - thanks for the clarification! You're right, I think some refactoring (and commenting) would really help here, but that's a separate issue. This is fine with me :)

Reply:
Thanks for sparing time for the review :)